starport/pkg/tarball/tarball.go

```diff
@@ -0,0 +1,65 @@
+package tarball
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"io"
+	"path/filepath"
+)
+
+var (
+	// ErrGzipFileNotFound the file not found in the gzip
+	ErrGzipFileNotFound = errors.New("file not found in the gzip")
+	// ErrInvalidGzipFile the gzip file is invalid
+	ErrInvalidGzipFile = errors.New("invalid gzip file")
+)
+
+// IsTarball checks if the file is a tarball
+func IsTarball(tarball []byte) error {
+	r, err := gzip.NewReader(bytes.NewReader(tarball))
+	if err == io.EOF || err == gzip.ErrHeader {
+		return ErrInvalidGzipFile
+	} else if err != nil {
+		return err
+	}
+	return r.Close()
+}
+
+// ReadFile founds and reads a specific file into a gzip file and folders recursively
+func ReadFile(tarball []byte, file string) ([]byte, error) {
```

Comment:
```suggestion
func ExtractFile(tarball io.Reader, out io.Writer, fileName string) (error) {
```

We should work with io.Reader/io.Writer's because these files might get huge.